src/render/recipe.rs

```diff
@@ -127,17 +126,28 @@ fn render_context(yaml_context: &serde_yaml::Mapping) -> BTreeMap<String, Value>
     context
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum RecipeRenderError {
+    #[error("Invalid YAML")]
+    InvalidYaml(#[from] serde_yaml::Error),
+
+    #[error(
+        "YAML does not follow regular recipe structure (map with build, requirements, outputs...)"
```

Comment:
This is not testing a certain schema right? The error can be a little more generic
```suggestion
        "Invalid Recipe File Format. The recipe file you've provided is not correctly formatted as a YAML mapping value. Please ensure your file adheres to the YAML standard and try again."
```